src/App.jsx

```diff
@@ -2,13 +2,46 @@ import "../src/assets/css/common.css";
 import "../src/assets/css/details.css";
 import "../src/assets/css/main.css";
 import "../src/assets/scss/common.scss";
-import Home from "./pages/Home";
+import Home from "./Home.jsx";
+import {useState} from "react";
 
+/**
+ * App Component
+ * Manages the application, including theme state and persistence.
+ *
+ * State:
+ * - theme: Tracks the current theme ("light" or "dark"), initialized from localStorage.
+ *
+ */
 function App() {
+    const [theme, setTheme] = useState(() => {
+        const savedTheme = localStorage.getItem("theme");
+
```

Comment:
You can extract all logic related to the localStorage into `storageService` that will expose methods to use, for example : 


```js
storageSession.getIemFromLocalStora(id)
```

etc...